csrc/python_frontend/python_bindings.cpp

```diff
@@ -171,6 +184,72 @@ std::vector<std::optional<bool>> computeContiguity(
   return contiguity;
 }
 
+std::tuple<std::vector<std::optional<bool>>, std::vector<int64_t>>
+computeTensorDescriptor(
+    const std::vector<int64_t>& sizes,
+    const std::vector<int64_t>& strides) {
+  NVF_CHECK(
+      sizes.size() == strides.size(),
+      "compute_contiguity: Sizes and strides must have the same number of dimensions");
+  std::vector<DimInfo> dim_info_vec;
+  for (auto i : c10::irange(sizes.size())) {
+    // NOTE: not supporting negative stride yet.
+    NVF_CHECK(strides[i] >= 0, "negative stride on tensor is not supported");
+    dim_info_vec.emplace_back(DimInfo{(int64_t)i, sizes[i], strides[i]});
+  }
+  // sort by stride
+  std::sort(
+      dim_info_vec.begin(),
+      dim_info_vec.end(),
+      [](const auto& l, const auto& r) { return l.stride > r.stride; });
+  // index to inner most dimension in sorted order.
+  int64_t last = (int64_t)sizes.size() - 1;
+  // Contiguity normallly is determined by the current dimension and one
+  // dimension to the right.  The innermost dimension, that is not broadcasted,
+  // does not have any dimension to it's right and needs to be specially marked
+  // contiguous.
+  while (last >= 0) {
+    // marking stride_order
+    dim_info_vec[last].stride_order = (int64_t)sizes.size() - 1 - last;
+    if (dim_info_vec[last].notBroadcast()) {
+      // setting current contiguity flag since it's not a broadcast dimension
+      dim_info_vec[last].contiguity = dim_info_vec[last].stride == 1;
```

Comment:
Can the loop below be merged into this loop by checking `dim_info_vec[last].stride == expect` instead of `== 1`?

Reply:
will do~